lightwood/data/timeseries_transform.py

```diff
@@ -236,6 +241,15 @@ def _ts_add_previous_target(df, target, nr_predictions, window, data_dtype, mode
         previous_target_values_arr.append(arr)
 
     df[f'__mdb_ts_previous_{target}'] = previous_target_values_arr
+    return df
+
+
+def _ts_add_future_target(df, target, nr_predictions, data_dtype, mode):
+    if target not in df:
+        return df
+    if data_dtype in (dtype.integer, dtype.float, dtype.array):
+        df[target] = df[target].astype(float)
```

Comment:
doesn't the casting already happen in the cleaner?

Reply:
Seems like it's not (may be some edge case), but I agree it's sloppy to do this here, will make sure it happens in the cleaner instead.